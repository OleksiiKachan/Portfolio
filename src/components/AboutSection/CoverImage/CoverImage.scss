@import '../../../assets/style/variables';
@import '../../../assets/style/keyframes';

.coverImage {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  background: url('https://res.cloudinary.com/oleksiikachan/image/upload/v1544127238/portfolio/images/header.png');
  background-size: cover;
  background-position: center;
  background-attachment: fixed;
  box-shadow: inset 2px 4px 4px rgba(0, 0, 0, 0.25);
  overflow: hidden;

  &__text {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-end;
    margin-right: 3rem;
  }

  &__title {
    padding: 1.3rem 0;
  }

  &__subtitle {
    padding: 1rem 0;
    font-size: 1.2rem;
  }

  &__resumeLink {
    color: $color-link;
    text-decoration: none;
    display: flex;
    align-items: flex-end;
    font-size: 1.2rem;
    font-weight: 700;
    margin: 1rem 0;

    &:hover {
      text-decoration: underline;
    }

    &::after {
      content: '';
      width: 12px;
      height: 12px;
      margin-left: 4px;
      background-size: contain;
      background-repeat: no-repeat;
      background-image: url('../../../assets/icons/icon_externalResource.svg');
    }
  }

  &__contacts {
    padding: 5rem 8rem;
    display: flex;
    flex-direction: column;
    background-image: linear-gradient(
      to right,
      rgba($color-surface, 0) 0%,
      rgba($color-surface, 0.7) 35%
    );
    transform: translateX(100%);
    animation: fromRightToCenter-noPop 700ms ease-in-out forwards;
  }

  &__contactsItem {
    margin: 0.5rem;
    transform: translateX(100%);
    opacity: 0;
    animation: opacity 400ms ease-in-out forwards,
      fromRightToCenter 400ms ease-in-out forwards;
  }
}

@media screen and (max-width: $breakpoint-tablet) {
  .coverImage {
    height: 25vh;

    &__contacts,
    &__text {
      display: none;
    }
  }
}
