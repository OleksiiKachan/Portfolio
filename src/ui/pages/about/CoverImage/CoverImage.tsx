import React from 'react';
import classNames from 'classnames';
import './CoverImage.scss';
import { SocialMediaButton } from '../../../elements';

/*
    <CoverImage
      name=''
      title=''
      resumeUrl=''
      contacts={}
      className=''
    />
*/

type PropsType = {
  name: string;
  title: string;
  resumeUrl: string;
  contacts: Array<{ type: string; url: string }>;
  className?: string;
};

export default ({ name, title, resumeUrl, contacts, className }: PropsType) => {
  return (
    <div className={classNames('coverImage', className)}>
      <div className={classNames('coverImage__text')}>
        <h1 className={classNames('coverImage__title')}>{name}</h1>
        <h2 className={classNames('coverImage__subtitle')}>{title}</h2>
        <a
          href={resumeUrl}
          target="_blank"
          rel="noopener noreferrer"
          className={classNames('coverImage__resumeLink')}
        >
          Printable CV
        </a>
      </div>
      <div className={classNames('coverImage__contacts')}>
        {contacts.map(item => {
          return (
            <SocialMediaButton
              key={contacts.indexOf(item)}
              type={item.type}
              url={item.url}
              className={classNames('coverImage__contactsItem')}
              style={{
                animationDelay: `${300 + contacts.indexOf(item) * 100}ms`,
              }}
            />
          );
        })}
      </div>
    </div>
  );
};
