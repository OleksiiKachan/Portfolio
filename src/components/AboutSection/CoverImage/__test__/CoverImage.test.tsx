import React from 'react';
import { shallow } from 'enzyme';
import CoverImage from '..';

describe('CoverImage', () => {
  it('Snapshot', () => {
    const component = shallow(
      <CoverImage
        name="name"
        title="title"
        contacts={[
          { type: 'type1', url: 'url1' },
          { type: 'type2', url: 'url2' },
        ]}
        resumeUrl="resume.url"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
