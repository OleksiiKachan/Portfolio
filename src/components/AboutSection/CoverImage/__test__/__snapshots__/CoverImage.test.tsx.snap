// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverImage Snapshot 1`] = `
<div
  className="coverImage"
>
  <div
    className="coverImage__text"
  >
    <h1
      className="coverImage__title"
    >
      name
    </h1>
    <h2
      className="coverImage__subtitle"
    >
      title
    </h2>
    <a
      className="coverImage__resumeLink"
      href="resume.url"
      target="_blank"
    >
      Printable CV
    </a>
  </div>
  <div
    className="coverImage__contacts"
  >
    <_default
      className="coverImage__contactsItem"
      key="0"
      type="type1"
      url="url1"
    />
    <_default
      className="coverImage__contactsItem"
      key="1"
      type="type2"
      url="url2"
    />
  </div>
</div>
`;
